app/src/main/java/sw19/moring03/paint/Views/DrawingView.java

```diff
@@ -115,6 +128,14 @@ private void selectTool() {
         }
     }
 
+    private Bitmap getCurrentBitmap() {
+        this.setDrawingCacheEnabled(true);
+        this.buildDrawingCache();
+        Bitmap bitmap = Bitmap.createBitmap(this.getDrawingCache());
+        this.setDrawingCacheEnabled(false);
```

Comment:
does it make sense to call setDrawingCacheEnabled true 2 times in this method??

Reply:
We actually need to do this. Calling `this.setDrawingCacheEnabled` first with `true` and then with `false` "re-builds" the drawing cache each time this code is executed. Otherwise, we will get the same result from `this.getDrawingCache()` even if something changed in the meantime (new tools's been drawn, etc.).

But I removed the occurrence of `this.setDrawingCacheEnabled` in the constructor. That was clearly a redundancy.

fixed with commit 7225edaa72.